### ISSUE TITLE
define.xml only validation (no datasets)

### ISSUE THREAD

**Issue by rradel** [cdisc-org/cdisc-rules-engine#603]
At this moment it is not possible to validate the define.xml only and only run rules related to define.xml. To run de validator, at this moment you should always provide datasets together with the define, but we have use cases where we just want to validate the define.xml. Also the possibility of running the define only will have a positive effect with regards to performance, run time and computer resource use.

**Comment by JozefAerts** [comment]
We have discussed this ...
Do you mean SDTM/ADaM rules that check "define-xml only"?
There are also very many Define-XML rules originating from the Define-XML team, and some of us have developed Schematron for this. However, the question arises whether we really start using Schematron within CORE ... Schematron only works on XML and when we gradually add/move to other formats for define (e.g. define.json), this wouldn't work.
But we can of course first start on rules from SDTM/ADaM that only use the define.xml and don't use a dataset itself.

**Comment by rradel** [comment]
indeed, I mean rules from SDTM/ADaM that only use the define.xml but don't use the datasets

**Comment by gerrycampion** [comment]
@JozefAerts @rradel 
Do you have an example rule from SDTM that uses the define.xml, but not datasets? I thought the SDTM rules only did crosschecks.

**Comment by rradel** [comment]
@gerrycampion 
I guess these will be mainly rules to check the define against the specifications like: 
- only variables that are referenced from a Dataset or Value level metadata should be included
- required elements must be included and cannot be empty
- cardinality and order check
- length attribute should be empty when DataType is not integer, float or text
- correct use of the extended value attribute
- check consistency between codelst DOMAIN and datasets defined
- check that empty permissible fields have a comment attached
- ... 

There are many more things that could be checked for define.xml only.
P21 also has an option to only validate defne.xml and uses these rules: https://www.pinnacle21.com/validation-rules/define-xml

**Comment by JozefAerts** [comment]
I will check a.s.a.p.

**Comment by JozefAerts** [comment]
For example, we have rule SEND4: "Variable-level metadata in Define-XML document should include the data Type (e.g., whether the variable value is a character or numeric)".

**Comment by gerrycampion** [comment]
@rradel I understand. I'm not sure of the current plan for the rules from the `Conformance Rules for Define-XML`


@JozefAerts 
> For example, we have rule SEND4: "Variable-level metadata in Define-XML document should include the data Type (e.g., whether the variable value is a character or numeric)".

The rule says:
> Variable-level metadata in Define-XML document uses the appropriate corresponding data type.

This sounds like a cross check against the study data. This may already be possible in the Engine using regex.

SEND271 provides a corresponding dataset-library crosscheck:
> The variable type in the .xpt dataset must match the SDTM specified data type.
